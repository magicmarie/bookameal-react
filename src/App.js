import React from "react";
import Notifications from "react-notify-toast";
import { Route, BrowserRouter, Switch } from "react-router-dom";
import Home from "./components/Home";
import SignUp from "./components/Signup";
import Login from "./components/Login";
import Navbar from "./components/common/Navbar";
import Footer from "./components/common/Footer";
import User from "./components/User";
import Admin from "./components/Admin";
import AuthRoute from "./components/common/Auth";
import AdminMenu from "./components/AdminMenu";
import UsersOrders from "./components/UsersOrders";

const App = ({ location }) => (
  <BrowserRouter>
    <div>
      <Navbar />
      <div className="container-fluid">
        <Notifications />
        <Switch>
          <Route path="/" exact component={Home} />
          <Route path="/signup" exact component={SignUp} />
          <Route path="/login" exact component={Login} />
          <AuthRoute path="/user" exact component={User} />
          <AuthRoute path="/admin" exact component={Admin} />
          <AuthRoute path="/adminmenu" exact component={AdminMenu} />
          <AuthRoute path="/usersorders" exact component={UsersOrders} />
        </Switch>
      </div>
      <Footer />
    </div>
  </BrowserRouter>
);

export default App;
