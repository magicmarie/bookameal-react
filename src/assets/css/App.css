body {
  font-family: sans-serif;
  font-size: 1rem;
  font-weight: 400;
  line-height: 1.5;
  color: #212529;
  text-align: left;
  background-image: url("../images/image1.jpeg");
  /* fixes image repeats */
  background-size: cover;
  background-repeat: no-repeat;
  height: 100%;
}

.text-center {
  text-align: center !important;
}

.button {
  font-weight: 400;
  text-align: center;
  white-space: nowrap;
  vertical-align: middle;
  user-select: none;
  border: 1px solid transparent;
  font-size: 1rem;
  line-height: 1.5;
  border-radius: 0.5rem;
  border-radius: 1.5rem;
  padding: 0.5rem 2rem;
  font-weight: 700;
  font-size: 1.25rem;
  color: #fff;
  margin: 40px;
}

.button:hover {
  outline: 0;
  cursor: pointer;
  box-shadow: 0 0 0 0.3rem rgba(0, 0, 0, 0.25);
}

.button.button-main {
  color: #fff;
  background-color: #7070da;
  border-color: #7070da;
}

.button.button-secondary {
  color: #7070da;
  border-color: #7070da;
}

.button .meal-list {
  margin: 5px;
  background-color: #eff4f7;
  border: #b9d7ea 1px solid;
  border-radius: 10px;
  font-size: 1.2rem;
}

.meal-button {
  margin-left: 10px;
}

/* Add style to the top navigation */

.topnav {
  background-color: #7070da;
  color: #fff;
  margin-bottom: 30px;
}

.topnav a:hover {
  background-color: #ddd;
  color: black;
}

h4 {
  font-weight: 900;
  font-size: 2.5rem;
}

.description {
  text-align: center;
  padding: 2rem;
  font-size: 22px;
}

.header {
  font-size: 1.25rem;
  font-weight: 700;
  display: block;
  padding: 1rem;
  font-weight: bold;
}

.menu {
  padding: 15px;
}

.menu-item .number {
  display: flex;
  align-items: center;
  justify-content: center;
  border: 2px solid #7070da;
  width: 2rem;
  height: 2rem;
  border-radius: 50%;
  vertical-align: middle;
  text-align: center;
}

.menu-item .price {
  padding: 0.25em 0.4em;
  font-size: 75%;
  font-weight: 700;
  line-height: 1;
  text-align: center;
  border-radius: 0.25rem;
  color: #fff;
  background-color: #7070da;
}

.menu-item .add {
  font-size: 2rem;
}

.menu-item {
  display: block;
  padding: 0.75rem 1.25rem;
  margin-bottom: 1px;
  background-color: #fff;
  font-size: 1.25rem;
  flex: 0 0 1;
  align-items: center;
  justify-content: space-between;
}

.centered {
  width: 50%;
  margin-left: 25%;
  margin-right: 25%;
  padding: 2px;
  display: flex;
  justify-content: space-around;
}

.form-signin-heading {
  margin-bottom: 30px;
  font-weight: normal;
  color: #007bff;
}

.footer {
  position: fixed;
  padding: 10px;
  left: 0;
  bottom: 0;
  width: 100%;
  background-color: #7070da;
  color: #fff;
  text-align: center;
}

.list-group-item {
  padding: 0 !important;
}

.meal {
  margin-right: 10px;
  background: #eee;
}

.mealcard {
  margin-right: 10px;
}
