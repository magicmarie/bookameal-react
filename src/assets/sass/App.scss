@import "bootstrap.min";

body {
  font-family: sans-serif;
  font-size: 1rem;
  font-weight: 400;
  line-height: 1.5;
  color: #212529;
  text-align: left;
  background-image: url("../images/image1.jpeg");
  /* fixes image repeats */
  background-size: cover;
  background-repeat: no-repeat;
  height: 100%;
}

.text-center {
  text-align: center !important;
}

.button {
  font-weight: 400;
  text-align: center;
  white-space: nowrap;
  vertical-align: middle;
  user-select: none;
  border: 1px solid transparent;
  font-size: 1rem;
  line-height: 1.5;
  border-radius: 0.5rem;
  border-radius: 1.5rem;
  padding: 0.5rem 2rem;
  font-weight: 700;
  font-size: 1.25rem;
  color: #fff;
  margin: 40px;
}

.button:hover {
  outline: 0;
  cursor: pointer;
  box-shadow: 0 0 0 0.3rem rgba(0, 0, 0, 0.25);
}

.button.button-main {
  color: #fff;
  background-color: #7070da;
  border-color: #7070da;
}

.button.button-secondary {
  color: #7070da;
  border-color: #7070da;
}

.button .meal-list {
  margin: 5px;
  background-color: #eff4f7;
  border: #b9d7ea 1px solid;
  border-radius: 10px;
  font-size: 1.2rem;
}

.meal-button {
  margin-left: 10px;
}

/* Add style to the top navigation */
.topnav {
  background-color: #7070da;
  margin-bottom: 15px;
}

.topnav a:hover {
  background-color: #ddd;
  color: black;
}

h4 {
  font-weight: 900;
  font-size: 2.5rem;
}

.description {
  text-align: center;
  padding: 2rem;
  font-size: 22px;
}

.header {
  font-size: 1.25rem;
  font-weight: 700;
  display: block;
  padding: 1rem;
  font-weight: bold;
}

.menu-item .add {
  color: #7070da;
  background-color: #cfcfd8;
}

.menu-item {
  padding: 1rem 1rem;
  margin-bottom: 1px;
  background-color: #ffffff96;
  flex: 0 0 1;
  align-items: center;
  justify-content: space-between;
  border-radius: 8px;
}

.menu-item :hover {
  background-color: #ffffff;
}

.centered {
  width: 50%;
  margin-left: 25%;
  margin-right: 25%;
  padding: 2px;
  display: flex;
  justify-content: space-around;
}

.form-signin-heading {
  margin-bottom: 30px;
  font-weight: normal;
  color: #007bff;
}

.footer {
  position: fixed;
  padding: 10px;
  left: 0;
  bottom: 0;
  width: 100%;
  background-color: #7070da;
  color: #fff;
  text-align: center;
}

.list-group-item {
  padding: 0 !important;
}

.meal {
  margin-right: 20px;
  background: #eee;
  margin-left: 10px;
  min-height: 8rem;
}

.mealcard {
  margin-left: 10px;
  margin-right: 10px;
}

.icon {
  font-size: 20px;
  color: #4978a7;
}

.row1 {
  margin: auto;
  align-items: center;
}

.vl {
  border-left: 2px solid #2b2b2b;
  height: 700px;
}

.menu {
  margin-bottom: 1rem;
}
