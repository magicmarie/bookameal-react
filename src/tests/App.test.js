import React from "react";
import ReactDOM from "react-dom";
import App from "../App";
import mockLocalStorage from "./MockLocalStorage";

global.localStorage = mockLocalStorage;

describe("App", () => {
  beforeEach(() => {
    Object.defineProperty(window, "localStorage", {
      value: mockLocalStorage
    });
  });

  afterEach(() => {
    // console.log(window.localStorage.getItem(""));
  });

  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
});
