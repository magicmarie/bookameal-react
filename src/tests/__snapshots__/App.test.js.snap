// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Navbar />,
        <div
          className="container-fluid"
>
          <_class
                    options={Object {}}
          />
          <Switch>
                    <Route
                              component={[Function]}
                              exact={true}
                              path="/"
                    />
                    <GuestRoute
                              component={[Function]}
                              exact={true}
                              path="/signup"
                    />
                    <GuestRoute
                              component={
                                        Object {
                                                  "$$typeof": Symbol(react.forward_ref),
                                                  "render": [Function],
                                                }
                              }
                              exact={true}
                              path="/login"
                    />
                    <AuthRoute
                              component={[Function]}
                              exact={true}
                              path="/user"
                    />
                    <AdminRoute
                              component={[Function]}
                              exact={true}
                              path="/admin"
                    />
                    <AdminRoute
                              component={[Function]}
                              exact={true}
                              path="/adminmenu"
                    />
                    <AuthRoute
                              component={[Function]}
                              exact={true}
                              path="/usersorders"
                    />
                    <AdminRoute
                              component={[Function]}
                              exact={true}
                              path="/adminorders"
                    />
                    <AuthRoute
                              component={[Function]}
                              exact={true}
                              path="/usercart"
                    />
                    <Route
                              component={[Function]}
                              exact={true}
                    />
          </Switch>
</div>,
        <Footer />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <_class
              options={Object {}}
/>,
            <Switch>
              <Route
                            component={[Function]}
                            exact={true}
                            path="/"
              />
              <GuestRoute
                            component={[Function]}
                            exact={true}
                            path="/signup"
              />
              <GuestRoute
                            component={
                                          Object {
                                                        "$$typeof": Symbol(react.forward_ref),
                                                        "render": [Function],
                                                      }
                            }
                            exact={true}
                            path="/login"
              />
              <AuthRoute
                            component={[Function]}
                            exact={true}
                            path="/user"
              />
              <AdminRoute
                            component={[Function]}
                            exact={true}
                            path="/admin"
              />
              <AdminRoute
                            component={[Function]}
                            exact={true}
                            path="/adminmenu"
              />
              <AuthRoute
                            component={[Function]}
                            exact={true}
                            path="/usersorders"
              />
              <AdminRoute
                            component={[Function]}
                            exact={true}
                            path="/adminorders"
              />
              <AuthRoute
                            component={[Function]}
                            exact={true}
                            path="/usercart"
              />
              <Route
                            component={[Function]}
                            exact={true}
              />
</Switch>,
          ],
          "className": "container-fluid",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "options": Object {},
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Route
                  component={[Function]}
                  exact={true}
                  path="/"
/>,
                <GuestRoute
                  component={[Function]}
                  exact={true}
                  path="/signup"
/>,
                <GuestRoute
                  component={
                                    Object {
                                                      "$$typeof": Symbol(react.forward_ref),
                                                      "render": [Function],
                                                    }
                  }
                  exact={true}
                  path="/login"
/>,
                <AuthRoute
                  component={[Function]}
                  exact={true}
                  path="/user"
/>,
                <AdminRoute
                  component={[Function]}
                  exact={true}
                  path="/admin"
/>,
                <AdminRoute
                  component={[Function]}
                  exact={true}
                  path="/adminmenu"
/>,
                <AuthRoute
                  component={[Function]}
                  exact={true}
                  path="/usersorders"
/>,
                <AdminRoute
                  component={[Function]}
                  exact={true}
                  path="/adminorders"
/>,
                <AuthRoute
                  component={[Function]}
                  exact={true}
                  path="/usercart"
/>,
                <Route
                  component={[Function]}
                  exact={true}
/>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/signup",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "component": Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "render": [Function],
                  },
                  "exact": true,
                  "path": "/login",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/user",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/admin",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/adminmenu",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/usersorders",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/adminorders",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/usercart",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Navbar />,
          <div
            className="container-fluid"
>
            <_class
                        options={Object {}}
            />
            <Switch>
                        <Route
                                    component={[Function]}
                                    exact={true}
                                    path="/"
                        />
                        <GuestRoute
                                    component={[Function]}
                                    exact={true}
                                    path="/signup"
                        />
                        <GuestRoute
                                    component={
                                                Object {
                                                            "$$typeof": Symbol(react.forward_ref),
                                                            "render": [Function],
                                                          }
                                    }
                                    exact={true}
                                    path="/login"
                        />
                        <AuthRoute
                                    component={[Function]}
                                    exact={true}
                                    path="/user"
                        />
                        <AdminRoute
                                    component={[Function]}
                                    exact={true}
                                    path="/admin"
                        />
                        <AdminRoute
                                    component={[Function]}
                                    exact={true}
                                    path="/adminmenu"
                        />
                        <AuthRoute
                                    component={[Function]}
                                    exact={true}
                                    path="/usersorders"
                        />
                        <AdminRoute
                                    component={[Function]}
                                    exact={true}
                                    path="/adminorders"
                        />
                        <AuthRoute
                                    component={[Function]}
                                    exact={true}
                                    path="/usercart"
                        />
                        <Route
                                    component={[Function]}
                                    exact={true}
                        />
            </Switch>
</div>,
          <Footer />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <_class
                options={Object {}}
/>,
              <Switch>
                <Route
                                component={[Function]}
                                exact={true}
                                path="/"
                />
                <GuestRoute
                                component={[Function]}
                                exact={true}
                                path="/signup"
                />
                <GuestRoute
                                component={
                                                Object {
                                                                "$$typeof": Symbol(react.forward_ref),
                                                                "render": [Function],
                                                              }
                                }
                                exact={true}
                                path="/login"
                />
                <AuthRoute
                                component={[Function]}
                                exact={true}
                                path="/user"
                />
                <AdminRoute
                                component={[Function]}
                                exact={true}
                                path="/admin"
                />
                <AdminRoute
                                component={[Function]}
                                exact={true}
                                path="/adminmenu"
                />
                <AuthRoute
                                component={[Function]}
                                exact={true}
                                path="/usersorders"
                />
                <AdminRoute
                                component={[Function]}
                                exact={true}
                                path="/adminorders"
                />
                <AuthRoute
                                component={[Function]}
                                exact={true}
                                path="/usercart"
                />
                <Route
                                component={[Function]}
                                exact={true}
                />
</Switch>,
            ],
            "className": "container-fluid",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "options": Object {},
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <Route
                    component={[Function]}
                    exact={true}
                    path="/"
/>,
                  <GuestRoute
                    component={[Function]}
                    exact={true}
                    path="/signup"
/>,
                  <GuestRoute
                    component={
                                        Object {
                                                            "$$typeof": Symbol(react.forward_ref),
                                                            "render": [Function],
                                                          }
                    }
                    exact={true}
                    path="/login"
/>,
                  <AuthRoute
                    component={[Function]}
                    exact={true}
                    path="/user"
/>,
                  <AdminRoute
                    component={[Function]}
                    exact={true}
                    path="/admin"
/>,
                  <AdminRoute
                    component={[Function]}
                    exact={true}
                    path="/adminmenu"
/>,
                  <AuthRoute
                    component={[Function]}
                    exact={true}
                    path="/usersorders"
/>,
                  <AdminRoute
                    component={[Function]}
                    exact={true}
                    path="/adminorders"
/>,
                  <AuthRoute
                    component={[Function]}
                    exact={true}
                    path="/usercart"
/>,
                  <Route
                    component={[Function]}
                    exact={true}
/>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "component": [Function],
                    "exact": true,
                    "path": "/",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "component": [Function],
                    "exact": true,
                    "path": "/signup",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "component": Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "render": [Function],
                    },
                    "exact": true,
                    "path": "/login",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "component": [Function],
                    "exact": true,
                    "path": "/user",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "component": [Function],
                    "exact": true,
                    "path": "/admin",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "component": [Function],
                    "exact": true,
                    "path": "/adminmenu",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "component": [Function],
                    "exact": true,
                    "path": "/usersorders",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "component": [Function],
                    "exact": true,
                    "path": "/adminorders",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "component": [Function],
                    "exact": true,
                    "path": "/usercart",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "component": [Function],
                    "exact": true,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
